x/dex/types/message_swap.go

```diff
@@ -47,5 +47,14 @@ func (msg *MsgSwap) ValidateBasic() error {
 	if err != nil {
 		return sdkerrors.Wrapf(sdkerrors.ErrInvalidAddress, "invalid creator address (%s)", err)
 	}
+
```

Comment:
I'm a little unsure why we added these checks here, as well call MustAccAddressFromBech32 for caller and receiver in msg_server. AFAIK it does the same thing. I would suggest removing the check from either within msg_server and having it check and return the addresses within validateBasic or or remove it from here.